python-backend/tests/factories.py

```diff
@@ -430,20 +428,16 @@ class Meta:
     email = factory.Faker('email')
     phone_no = factory.Faker('numerify', text='###-###-####')
     last_logon = TODAY
-    idir_user_detail = factory.SubFactory('tests.factories.IdirUserDetailFactory')
-
-    @factory.post_generation
-    def idir_user_detail(obj, create, extracted, **kwargs):
-        if not create:
-            return
-
-        IdirUserDetailFactory.create(core_user_id=obj.core_user_id, **kwargs)
+    idir_user_detail = factory.RelatedFactory('tests.factories.IdirUserDetailFactory', 'core_user')
 
 
 class IdirUserDetailFactory(BaseFactory):
     class Meta:
         model = IdirUserDetail
 
+    class Params:
+        core_user = factory.SubFactory('tests.factories.CoreUserFactory')
```

Comment:
You only need to reference factories like this if they are declared later in the file, as the binding happens when the file is loaded. CoreUserFactory is declared already, so this line can just be `core_user = factory.SubFactory(CoreUserFactory)`

Reply:
Not a big deal, however. This is more for a heads-up on slightly nicer looking style; it doesn't change the functionality at all.